src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs

```diff
@@ -0,0 +1,4224 @@
+            {
+                public static CancellationTokenSource CreateLinkedTokenSource(CancellationToken token);
+                public static CancellationTokenSource CreateLinkedTokenSource(CancellationToken token1, CancellationToken token2);
+                public static CancellationTokenSource CreateLinkedTokenSource(params CancellationToken[] tokens);
+            }
+            */
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(CancellationToken token){}
+    public static void Method(params CancellationToken[] tokens){}
+    public static void Method(CancellationToken token1, CancellationToken token2){}
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_CancellationTokenSource_ParamsUsed_Order3()
+        {
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(CancellationToken token1, CancellationToken token2){}
+    public static void Method(params CancellationToken[] tokens){}
+    public static void Method(CancellationToken token){}
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_CancellationTokenSource_ParamsUsed_Order4()
+        {
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(CancellationToken token1, CancellationToken token2){}
+    public static void Method(CancellationToken token){}
+    public static void Method(params CancellationToken[] tokens){}
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_CancellationTokenSource_ParamsUsed_Order5()
+        {
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(params CancellationToken[] tokens){}
+    public static void Method(CancellationToken token){}
+    public static void Method(CancellationToken token1, CancellationToken token2){}
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_CancellationTokenSource_ParamsUsed_Order6()
+        {
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(params CancellationToken[] tokens){}
+    public static void Method(CancellationToken token1, CancellationToken token2){}
+    public static void Method(CancellationToken token){}
```

Comment:
Can we get rid of all these duplicate tests? I don't think these are testing different code paths.